src/rosflight_joy/rc_keyboard

```diff
@@ -4,56 +4,99 @@
+        self.auto_arm = rospy.get_param('~auto_arm', False)
+        self.auto_takeoff = rospy.get_param('~auto_takeoff', False)
+        if not self.auto_arm:
+            self.auto_takeoff = False
+            self.init = True
+        self.rate = rospy.Rate(30)
+        if self.auto_arm:
+            rospy.logwarn("[rc_keyboard] AUTO-ARM is ON. Waiting for firmware version")
+        else:
+            rospy.loginfo("[rc_keyboard] AUTO-ARM is OFF.")
+        if self.auto_takeoff:
+            rospy.logwarn("[rc_keyboard] AUTO-TAKEOFF is ON.")
+        else:
+            rospy.loginfo("[rc_keyboard] AUTO-TAKEOFF is OFF.")
+    def run(self):
+        while not rospy.is_shutdown():
+            self.rate.sleep()
           
+            if self.got_version:
+                if self.auto_arm and not self.init:
+                    self.arm()
+                    if self.is_armed:
+                        self.values['aux1'] = -1
+                        if self.auto_takeoff:
+                            self.values['F'] = 1
+                        self.init = True
+                        rospy.loginfo("[rc_keyboard] auto-arm was successful")
+                        if not self.auto_takeoff:
+                            rospy.loginfo("[rc_keyboard] increase throttle with 'w' key for manual takeoff")
+            else:
+                continue
+
+            if self.init:
+                self.update()
+
             msg = RCRaw()
-            msg.header.stamp = rospy.Time.now()
-            msg.values[0] = self.get_value('x') 
-            msg.values[1] = self.get_value('y') 
-            msg.values[2] = self.get_value('F') 
-            msg.values[3] = self.get_value('z') 
-            msg.values[4] = self.get_value('aux1')
-            msg.values[5] = self.get_value('aux2')
+            msg.header.stamp = rospy.Time.now()    # key: action
+            msg.values[0] = self.get_value('x')    # left/right: roll
+            msg.values[1] = self.get_value('y')    # up/down: pitch
+            msg.values[2] = self.get_value('F')    # w/s: throttle
+            msg.values[3] = self.get_value('z')    # a/d: yaw
+            msg.values[4] = self.get_value('aux1') # o: rc override toggle
+            msg.values[5] = self.get_value('aux2') # m: arm toggle (assumes ARM_CHANNEL=4)
             msg.values[6] = self.get_value('aux3')
             msg.values[7] = self.get_value('aux4')
 
-            rc_pub.publish(msg)
-            
-            rospy.sleep(0.02)
+            self.rc_pub.publish(msg)
+            self.rate.sleep()
 
         self.quit()
 
 
     def get_value(self, key):
         return int(self.values[key] * 500 + 1500)
 
+    def arm(self):
+        self.values['aux2'] = 1  # arm toggle ON
+        self.values['aux1'] = 1  # rc override ON
+        self.values['F'] = -1    # throttle to 0%
+
+        # In case ARM_CHANNEL is not set, this will be required to arm
+        self.values['z'] = 1     # yaw at 100%
```

Comment:
So this gets reset to zero by the `slide_to_zero` functionality? I'd assume this means that the multirotor would yaw slightly at first if auto takeoff is enabled, but I guess it works well enough? Or maybe I'm just missing something

Reply:
Great catch - the idea was to set yaw back to zero once armed, but I neglected to do that. It was just going to zero by the `slide_to_zero` function, as you said. I fixed it in 31d9c4f.